src/components/ComposerQuill/composer.js

```diff
@@ -113,20 +146,33 @@ class Composer extends React.Component {
 
   // Goes through the list and checks if the search term is in it
   handleMention(searchTerm, renderList) {
+    const participants = this.props.mentions.participants.current;
+
     if (searchTerm.length === 0) {
-      renderList(list, searchTerm);
+      renderList(participants, searchTerm);
     } else {
       const matches = [];
 
-      for (let i = 0; i < list.length; i += 1) {
-        if (list[i].value.toLowerCase().indexOf(searchTerm.toLowerCase()) >= 0) {
-          matches.push(list[i]);
+      for (let i = 0; i < participants.length; i += 1) {
+        if (participants[i].displayName.toLowerCase().indexOf(searchTerm.toLowerCase()) >= 0) {
+          matches.push(participants[i]);
         }
       }
       renderList(matches, searchTerm);
     }
   }
 
+  handleMentionSelect(item, insertItem) {
+    const copy = {...item};
+    const name = item.displayName;
+    const index = name.indexOf(' ');
+
+    // get the first name if there is one, otherwise copy the name
+    copy.value = index > 0 ? name.substring(0, index) : name;
```

Comment:
I think it shows the first name unless there are two people in the space with the same first name.

Reply:
Oooh, that's fancy. I'll add that too